src/btree/bt_page.c

```diff
@@ -15,6 +15,434 @@ static int __inmem_row_int(WT_SESSION_IMPL *, WT_PAGE *, size_t *);
 static int __inmem_row_leaf(WT_SESSION_IMPL *, WT_PAGE *, bool *);
 static int __inmem_row_leaf_entries(WT_SESSION_IMPL *, const WT_PAGE_HEADER *, uint32_t *);
 
+/*
+ * __page_build_ref --
+ *     Create a ref from a base image or a delta.
+ */
+static int
+__page_build_ref(WT_SESSION_IMPL *session, WT_REF *parent_ref, WT_CELL_UNPACK_ADDR *base_key,
+  WT_CELL_UNPACK_ADDR *base_val, WT_CELL_UNPACK_DELTA_INT *delta, bool base, WT_REF **refp,
+  size_t *incrp)
+{
+    WT_REF *ref;
+    uint8_t key_type, value_type;
+
+    WT_ASSERT(session, incrp != NULL);
+
+    WT_RET(__wt_calloc_one(session, refp));
+    *incrp += sizeof(WT_REF);
+
+    ref = *refp;
+    ref->home = parent_ref->page;
+    key_type = base ? base_key->type : delta->key.type;
+    value_type = base ? base_val->type : delta->value.type;
+
+    switch (key_type) {
+    case WT_CELL_KEY:
+        if (base)
+            __wt_ref_key_onpage_set(parent_ref->page, ref, base_key);
+        else {
+            WT_RET(__wti_row_ikey(session, 0, delta->key.data, delta->key.size, ref));
+            *incrp += sizeof(WT_IKEY) + delta->key.size;
+        }
+        break;
+    case WT_CELL_KEY_OVFL:
+    /* Overflow keys are not supported. */
+    default:
+        WT_RET(__wt_illegal_value(session, delta->key.type));
+    }
+
+    switch (value_type) {
+    case WT_CELL_ADDR_INT:
+        F_SET(ref, WT_REF_FLAG_INTERNAL);
+        break;
+    case WT_CELL_ADDR_LEAF:
+    case WT_CELL_ADDR_LEAF_NO:
+        F_SET(ref, WT_REF_FLAG_LEAF);
+        break;
+    case WT_CELL_ADDR_DEL:
+    /* Fast truncated pages are not supported. */
```

Comment:
Can you expand the comment here?

Reply:
`Fast truncated pages are not supported when constructing an internal page with deltas`?

Reply:
yes, based on what we discussed and observed yesterday.